PrimeNumbers/ProcessorInfo.h

```diff
@@ -92,37 +92,56 @@ void GetProcessorInfo(int* processorCount, int* processorGroupCount)
 /// <param name="processorCount"></param>
 /// <param name="isMultiCpuGroup"></param>
 /// <param name="threadHandles"></param>
-void SetThreadAffinity(int processorCount, bool isMultiCpuGroup, std::vector<HANDLE>& threadHandles)
+void SetThreadAffinity(int processorCount, 
+					   int numGroups, 
+	                   std::vector<HANDLE>& threadHandles, 
+	                   bool affinitizeEveryOtherProc)
 {
 	assert(threadHandles.size() == processorCount);
 
+	bool isMultiCpuGroup = (numGroups > 1);
+
 	for (int procNo = 0; procNo < processorCount; procNo++)
 	{
-		GroupProcNo groupProcNo(procNo);
+		int adjustedProcNo = (affinitizeEveryOtherProc ? (procNo * 2) : procNo);
```

Comment:
When `affinitizeEveryOtherProc = true`, shouldn't the loop be up to `processorCount / 2`? What does it mean to have `procNo * 2` which exceeds the value of `processorCount`?

Reply:
you cannot specify a processorCount that's more than (total (logical) processors / 2) so if you have 256 logical procs, you can specify max 128 for processorCount. so procNo 0 will be affi to logical proc 0, procNo 1 will be affi to logic proc 2 and so on.